io/src/real_sense_2_grabber.cpp

```diff
@@ -0,0 +1,394 @@
+
+    auto vertices_ptr = points.get_vertices ();
+
+#ifdef _OPENMP
+#pragma omp parallel for 
+#endif
+    for (int index = 0; index < cloud->points.size (); ++index)
+    {
+      auto ptr = vertices_ptr + index;
+      auto& p = cloud->points[index];
+
+      p.x = ptr->x;
+      p.y = ptr->y;
+      p.z = ptr->z;
+    }
+
+    return cloud;
+  }
+
+  pcl::PointCloud<pcl::PointXYZI>::Ptr 
+  RealSense2Grabber::convertInfraredDepthToPointXYZI (const rs2::points & points, rs2::video_frame & ir)
+  {
+    pcl::PointCloud<pcl::PointXYZI>::Ptr cloud (new pcl::PointCloud<pcl::PointXYZI> ());
+
+    auto sp = points.get_profile ().as<rs2::video_stream_profile> ();
+    cloud->width = sp.width ();
+    cloud->height = sp.height ();
+    cloud->is_dense = false;
+    cloud->points.resize (points.size ());
+
+    auto vertices_ptr = points.get_vertices ();
+    auto texture_ptr = points.get_texture_coordinates ();
+
+    uint8_t r, g, b;
+
+    if (ir_format_ == RS2_FORMAT_UYVY)
+    {
+#ifdef _OPENMP
+#pragma omp parallel for 
+#endif
+      for (int index = 0; index < cloud->points.size (); ++index)
+      {
+        auto ptr = vertices_ptr + index;
+        auto uvptr = texture_ptr + index;
+        auto& p = cloud->points[index];
+
+        p.x = ptr->x;
+        p.y = ptr->y;
+        p.z = ptr->z;
+
+        std::tie (r, g, b) = getTextureColor (ir, uvptr->u, uvptr->v);
+
+        p.intensity = 0.299f * static_cast <float> (r) + 0.587f * static_cast <float> (g) + 0.114f * static_cast <float> (b);
+
+      }
+    }
+    else
+    {
+#ifdef _OPENMP
+#pragma omp parallel for 
+#endif
+      for (int index = 0; index < cloud->points.size (); ++index)
+      {
+        auto ptr = vertices_ptr + index;
+        auto uvptr = texture_ptr + index;
+        auto& p = cloud->points[index];
+
+        p.x = ptr->x;
+        p.y = ptr->y;
+        p.z = ptr->z;
+
+        p.intensity = getTextureIntensity(ir, uvptr->u, uvptr->v);
+
+      }
+    }
+
+
+
+    return cloud;
+  }
+
+  pcl::PointCloud<pcl::PointXYZRGB>::Ptr 
+  RealSense2Grabber::convertRGBDepthToPointXYZRGB (const rs2::points & points, rs2::video_frame & rgb)
+  {
+    pcl::PointCloud<pcl::PointXYZRGB>::Ptr cloud (new pcl::PointCloud<pcl::PointXYZRGB> ());
+
+    auto sp = points.get_profile ().as<rs2::video_stream_profile> ();
+    cloud->width = sp.width ();
+    cloud->height = sp.height ();
+    cloud->is_dense = false;
+    cloud->points.resize (points.size ());
+
+    auto vertices_ptr = points.get_vertices ();
+    auto texture_ptr = points.get_texture_coordinates ();
+
+#ifdef _OPENMP
+#pragma omp parallel for 
+#endif
+    for (int index = 0; index < cloud->points.size (); ++index)
+    {
+      auto ptr = vertices_ptr + index;
+      auto uvptr = texture_ptr + index;
+      auto& p = cloud->points[index];
+
+      p.x = ptr->x;
+      p.y = ptr->y;
+      p.z = ptr->z;
+
+      std::tie (p.r, p.g, p.b) = getTextureColor (rgb, uvptr->u, uvptr->v);
+
+      p.a = 255;
```

Comment:
'PointRGB` is already initialized with alpha 255.

Reply:
I added this because in testing I found that in testing the alpha was initialized to zero.  I'll go back and double check again.

Reply:
https://github.com/PointCloudLibrary/pcl/blob/fc88d45bcc4f49904bcff1b4359f1310ab516292/common/include/pcl/impl/point_types.hpp#L619